services/workflows-service/src/auth/local/local-auth.guard.ts

```diff
@@ -1,26 +1,21 @@
-import { Injectable, ExecutionContext } from '@nestjs/common';
+import { CanActivate, Injectable, ExecutionContext } from '@nestjs/common';
 import { AuthGuard } from '@nestjs/passport';
 import { SupabaseService } from '../../supabase/supabase.service';
 
 @Injectable()
-export class LocalAuthGuard extends AuthGuard('local') {
+export class LocalAuthGuard extends AuthGuard('local') implements CanActivate {
   constructor(private readonly supabaseService: SupabaseService) {
     super();
   }
 
-  async canActivate(context: ExecutionContext): Promise<boolean> {
-    const result = (await super.canActivate(context)) as boolean;
+  canActivate(context: ExecutionContext): Promise<boolean> {
+    const result = super.canActivate(context) as boolean;
     const request = context.switchToHttp().getRequest();
-
-    // Use the injected service
     if (result && request.user) {
       const fullUrl = request.protocol + '://' + request.get('host') + request.originalUrl;
-      await this.supabaseService.logSignIn(fullUrl);
+      this.supabaseService.logSignIn(fullUrl);
```

Comment:
in case this function will throw an error it will be unhandled error

Add logger in constructor:
```
  protected readonly logger: AppLoggerService,
```

so we have to use `.catch` and log in case error has been raised:
```
.catch(error => {
           this.logger.error(`Error logging sign-in`, {
        error,
      });
     console.error('Error logging sign-in:', error); // Handle error without affecting the flow
 });
 ```

Reply:
we are already doing a try catch in this logSignIn [here](https://github.com/ballerine-io/ballerine/pull/2363/commits/e4ec6f3e9129148357f71d463d6042f2b338f1f3#diff-f8ac5ea13f525fbf35199da5604438fa51feb1089cd8986c59ae52d87375ebd6R48)

should we still have to write try catch again ?